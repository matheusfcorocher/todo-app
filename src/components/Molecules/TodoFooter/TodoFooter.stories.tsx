import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import TodoFooter from './TodoFooter';

export default {
  title: 'Molecules/TodoFooter',
  component: TodoFooter,
  decorators: [
    (Story) => (
      <ul style={{ listStyleType: 'none', margin: 0, padding: 0 }}>
        <Story />
      </ul>
    ),
  ]
} as ComponentMeta<typeof TodoFooter>;

const Template: ComponentStory<typeof TodoFooter> = (args) => <TodoFooter {...args} />;

export const Default = Template.bind({});
Default.args = {
};