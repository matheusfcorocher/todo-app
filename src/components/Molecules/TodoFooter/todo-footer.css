body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.todo-menu {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
}

.todo-count {
  text-align: left;
  float: left;
}