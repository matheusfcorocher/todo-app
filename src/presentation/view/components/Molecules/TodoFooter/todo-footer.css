@import '../../../styles/colors/colors.css';
@import '../../../styles/round-corners/round-corners.css';
@import '../../../styles/spacing/spacing.css';

.todo-footer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
}

.todo-count {
  text-align: left;
  float: left;
  color: var(--gray-color)
}

.todo-filters {
  display: flex;
  list-style: none;
  padding: 0rem;
  gap: var(--spacing-sm);
}

.todo-filter {
  width: 5rem;
  padding: var(--spacing-xs);
  border-radius: var(--round-corner-sm);
  text-decoration: none;
  color: var(--tertiary-color);
  border: none;
  text-decoration: none;
}

.active {
  background-color: var(--tertiary-color);
  color: var(--white-color);
}


@media (max-width: 30rem) {
  .todo-footer {
    flex-direction: column;
  }
  
  .todo-count {
    text-align: center;
    float: none;
  }
}