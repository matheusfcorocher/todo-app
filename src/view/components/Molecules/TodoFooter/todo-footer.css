@import '../../../styles/colors/colors.css';
@import '../../../styles/round-corners/round-corners.css';
@import '../../../styles/spacing/spacing.css';

.todo-footer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
}

.todo-count {
  text-align: left;
  float: left;
}

.todo-filters {
  display: flex;
  list-style: none;
  padding: 0rem;
  gap: var(--spacing-xl);
}

.todo-filter {
  width: 5rem;
  padding: var(--spacing-xs);
  border-radius: var(--round-corner-sm);
  text-decoration: none;
  color: var(--tertiary-color);
  border: none;
  text-decoration: none;
}

.active {
  background-color: var(--tertiary-color);
  color: var(--white-color);
}
