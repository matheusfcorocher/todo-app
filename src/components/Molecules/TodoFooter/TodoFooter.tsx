import React, { useState } from 'react';
import '../../../App.css';
import './todo-footer.css';

interface TodoFooter {

}

function TodoFooter({}: TodoFooter) {
    return (
        <div role="group" className={`todo-footer`}>
            <span className={`todo-count`}>
                <strong>1&nbsp;</strong>
                item left
            </span>
            <ul className={`todo-filters`}>
                <li className={'todo-filter'}>All</li>
                <li className={'todo-filter'}>Active</li>
                <li className={'todo-filter'}>Completed</li>
            </ul>
        </div>
    );
}

export default TodoFooter;
