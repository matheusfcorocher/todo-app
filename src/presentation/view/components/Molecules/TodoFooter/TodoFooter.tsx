import React, { useState } from "react";
import { IsCompletedFilterControllerReturnType } from "../../../../controllers/IsCompletedFilterController";
import { TodoTaskControllerReturnType } from "../../../../controllers/TodoTaskController";
import Button from "../../Atoms/Button/Button";
import "./todo-footer.css";

interface TodoFooter {
  isCompletedFilterController: IsCompletedFilterControllerReturnType;
  todoTaskController: TodoTaskControllerReturnType;
}

function TodoFooter({ todoTaskController, isCompletedFilterController }: TodoFooter) {
  const {
    handleDeleteAllCompletedTodoTasks,
    getOnlyActiveTodoTasks,
    getIsThereAnyTodoTaskCompleted,
  } = todoTaskController;

  const {handleChangeFilter, getIsCompletedFilter} = isCompletedFilterController;
  const filter = getIsCompletedFilter();
  const handleClick = (filter?: boolean): void => {
    handleChangeFilter(filter);
  };

  return (
    <div role="group" className={`todo-footer`}>
      <span className={`todo-count`}>
        <strong>{getOnlyActiveTodoTasks().length}&nbsp;</strong>
        item left
      </span>
      <ul className={`todo-filters`}>
        <li>
          <a
            title={"Filter all todo tasks"}
            className={`todo-filter ${filter == undefined ? "active" : ""}`}
            onClick={(e) => handleClick(undefined)}
            href="#/"
          >
            All
          </a>
        </li>
        <li>
          <a
            title={"Filter active todo tasks"}
            className={`todo-filter ${filter == false ? "active" : ""}`}
            onClick={(e) => handleClick(false)}
            href="#/active"
          >
            Active
          </a>
        </li>
        <li>
          <a
            title={"Filter completed todo tasks"}
            className={`todo-filter ${filter == true ? "active" : ""}`}
            onClick={(e) => handleClick(true)}
            href="#/completed"
          >
            Completed
          </a>
        </li>
      </ul>
      {getIsThereAnyTodoTaskCompleted() === true && (
        <Button
          handleFunction={handleDeleteAllCompletedTodoTasks}
          title={"Clear Completed"}
          variant={"primary"}
          size={"small"}
          fontColor={"gray"}
        />
      )}
    </div>
  );
}

export default TodoFooter;
