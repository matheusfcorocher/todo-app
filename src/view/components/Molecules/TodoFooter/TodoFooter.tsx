import React, { useState } from 'react';
import { HandleDeleteAllCompletedTodoTasks, HandleFilter } from '../../../../App';
import Button from '../../Atoms/Button/Button';
import './todo-footer.css';

interface TodoFooter {
    areThereTodoTasksCompleted: boolean;
    todosQuantity: number;
    handleFilter: HandleFilter;
    handleDeleteAllCompletedTodoTasks: HandleDeleteAllCompletedTodoTasks;
}

function TodoFooter({ todosQuantity = 0, areThereTodoTasksCompleted = false, handleFilter, handleDeleteAllCompletedTodoTasks }: TodoFooter) {
    const handleClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, filter?: boolean): void => {
        document.querySelectorAll('.active').forEach(e => e.classList.remove('active'));
        event.currentTarget.classList.toggle('active')
        handleFilter(filter);
    }

    return (
        <div role="group" className={`todo-footer`}>
            <span className={`todo-count`}>
                <strong>{todosQuantity}&nbsp;</strong>
                item left
            </span>
            <ul className={`todo-filters`}>
                <li><a title={"Filter all todo tasks"} className={'todo-filter active'} onClick={(e) => handleClick(e, undefined)} href="#/">All</a></li>
                <li><a title={"Filter active todo tasks"} className={'todo-filter'} onClick={(e) => handleClick(e, false)} href="#/active">Active</a></li>
                <li><a title={"Filter completed todo tasks"} className={'todo-filter'} onClick={(e) => handleClick(e, true)} href="#/completed">Completed</a></li>
            </ul>
            {
                areThereTodoTasksCompleted === true &&
                <Button
                    handleFunction={handleDeleteAllCompletedTodoTasks}
                    title={"Clear Completed"}
                    variant={"primary"}
                    size={"medium"}
                    fontColor={"gray"}
                />
            }
        </div>
    );
}

export default TodoFooter;
