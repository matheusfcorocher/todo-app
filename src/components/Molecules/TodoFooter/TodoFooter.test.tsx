import React from 'react';
import { render, screen } from '@testing-library/react';
import TodoFooter from './TodoFooter';

describe("::Components ::Molecules ::TodoFooter", () => {
  test('renders TodoFooter', () => {
      render(<TodoFooter />);
      const todoFooter = screen.getByRole("group");
      expect(todoFooter).toMatchSnapshot();
  })
})