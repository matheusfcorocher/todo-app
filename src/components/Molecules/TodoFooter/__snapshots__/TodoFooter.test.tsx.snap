// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`::Components ::Molecules ::TodoFooter renders TodoFooter 1`] = `
<div
  class="todo-footer"
  role="group"
>
  <span
    class="todo-count"
  >
    <strong>
      0
      Â 
    </strong>
    item left
  </span>
  <ul
    class="todo-filters"
  >
    <li>
      <a
        class="todo-filter active"
        href="#/"
      >
        All
      </a>
    </li>
    <li>
      <a
        class="todo-filter"
        href="#/active"
      >
        Active
      </a>
    </li>
    <li>
      <a
        class="todo-filter"
        href="#/completed"
      >
        Completed
      </a>
    </li>
  </ul>
</div>
`;
